src/main/appender.cpp

```diff
@@ -373,6 +373,13 @@ void BaseAppender::Append(Value value) { // NOLINT: template stuff
 	AppendValue(value);
 }
 
+void BaseAppender::Append(DataChunk &target, Value value, idx_t column) {
+	if (column >= target.ColumnCount()) {
+		throw InvalidInputException("Too many appends for chunk!");
+	}
+	target.SetValue(column, target.size(), value);
```

Comment:
This approach sets the default value in `column` for row `target.size()`, so it appends a single default value to the data chunk. It seems inefficient and tailored to implementations that set the data chunk size after each row and use a row-based approach.

I am skeptical because this function feels like a helper function for specific workloads with a few sparse values. In a general approach, a driver would probably fall back to the (for bulk workloads) more efficient `add_column/clear_columns` approach.

If we decide to add this, I think we should change the semantics to include a `row_idx`. That way, filling the data chunk in any order (also column-at-a-time) is possible, and eventually, a few sparse default values can be set. Additionally, it allows the data chunk size to be set once at the end.

Before adding this, remember that we are trying to keep the C API slim and have a skeptical take on what we are adding. I am not saying we should not; I want to ensure it adds enough value and is not a helper function for a less significant use case. Linking this PR for context: https://github.com/duckdb/duckdb/pull/14992.

Reply:
Instead, we could add a `duckdb_append_value` function that aligns with `duckdb_append_default`. This approach would extend the row-based append functionality to include any types that can be a `Value`.

Reply:
You are right, I didn't notice the issue with the size. I can update the function definition to include row index as a parameter to avoid having the need to set the chunk size.

As for the `duckdb_append_value` this was already merged: https://github.com/duckdb/duckdb/pull/15065

Regarding keeping the C API slim, I agree that the API should be slim, but in my opinion it is already confusing that there are already different ways to achieve the same. For example, we can bind to prepare statements with `duckdb_bind_value` or one of the specialized `duckdb_bind_xxx` functions. We can write to appender with data chunk or with rows based functions. Having different ways to accomplish the same thing is confusing and raises questions for those who are new to the API.

Reply:
> You are right, I didn't notice the issue with the size. I can update the function definition to include row index as a parameter to avoid having the need to set the chunk size.

Great 👍 

> Regarding keeping the C API slim, I agree that the API should be slim, but in my opinion it is already confusing that there are already different ways to achieve the same.

Yeah, we are already trying to tackle this. However, we just decided to keep everything that is not tagged for deprecation stable. I don't know if that is an argument to add more duplication. But given that @Mytherin is okay with adding this function, I'd say you can finish up this PR (we need tests, etc.). Then, I'll review it again. :)